src/dispatcher/server.rs

```diff
@@ -148,40 +162,84 @@ impl SignatureDispatcher for Ed25519SignatureDispatcher {
 }
 
 #[tokio::main]
-async fn main() -> Result<(), Box<dyn std::error::Error>> {
+async fn main() -> remote_signer::Result<()> {
```

Comment:
Can you put this function at the bottom of the file? I think it is more orderly that way.

Reply:
I would actually put it on top if we change order :-P 

Because when I want to understand what a program does I always look for `main`, and then I dig

Reply:
if we decide to move, I would rather it will be in the beginning...

Reply:
`main` has been always historically at the bottom though: the times when compilers need to parse the file in order and a function could not call something declared below but only something above!